scripts/SasNetworkGuard.psm1

```
@@ -0,0 +1,184 @@
+    $script:SasNetworkGuardLastWiredEvidence = 'none'
+    $config = Get-SasNetworkGuardConfig
+    if ($null -eq $config) {
+        $script:SasNetworkGuardLastWiredEvidence = "config_error:$script:SasNetworkGuardConfigError"
+        return $false
+    }
+    $hasAny = @($config.allowedDnsSuffixes + $config.allowedWindowsDomains + $config.allowedLocalIpCidrs + $config.allowedGatewayCidrs + $config.allowedDnsServerCidrs).Count -gt 0
+    if (-not $hasAny) { return $false }
+    $text = if ($null -eq $NetworkText) { '' } else { $NetworkText }
+    $lower = $text.ToLowerInvariant()
+    foreach ($suffix in $config.allowedDnsSuffixes) {
+        if ($lower.Contains($suffix.ToLowerInvariant())) { $script:SasNetworkGuardLastWiredEvidence = "dns_suffix=$suffix"; return $true }
```

PSScriptAnalyzer (1.25.0)

[info] The cmdlet 'Test-SasNorthwellWiredEvidence' returns an object of type 'System.Boolean' but this type is not declared in the OutputType attribute.

(PSUseOutputTypeCorrectly)